reactive/kubernetes_master.py

```diff
@@ -1720,19 +1721,35 @@ def configure_scheduler():
 
 def setup_basic_auth(password=None, username='admin', uid='admin',
                      groups=None):
-    '''Create the htacces file and the tokens.'''
+    '''Add or update an entry in /root/cdk/basic_auth.csv.'''
     root_cdk = '/root/cdk'
     if not os.path.isdir(root_cdk):
         os.makedirs(root_cdk)
     htaccess = os.path.join(root_cdk, 'basic_auth.csv')
     if not password:
         password = token_generator()
-    with open(htaccess, 'w') as stream:
-        if groups:
-            stream.write('{0},{1},{2},"{3}"'.format(password,
-                                                    username, uid, groups))
-        else:
-            stream.write('{0},{1},{2}'.format(password, username, uid))
+
+    newline = '{0},{1},{2}'.format(password, username, uid)
+    if groups:
+        newline += ',"{0}"'.format(groups)
+
+    # If we have an existing line for this username, update it, leaving
+    # lines for other users unchanged.
+    with fileinput.input(htaccess, inplace=True) as f:
+        user_written = False
```

Comment:
Oh, and the name of this variable confused me for a second.  I might suggest `wrote_user_line` instead.

Reply:
Thanks, updated.